src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.hpp

```diff
@@ -129,4 +129,234 @@ tnsr::ii<DataType, SpatialDim, Frame> extrinsic_curvature(
+tnsr::i<DataType, SpatialDim, Frame> spatial_deriv_of_lapse(
+    const Scalar<DataType>& lapse,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes time derivative of lapse (N) from the generalized
+ *        harmonic variables, lapse, shift and the spacetime unit normal 1-form.
+ *
+ * \details Let the generalized harmonic conjugate momentum and spatial
+ * derivative variables be \f$\pi_{ab} = -n^c \partial_c \psi_{ab} \f$ and
+ * \f$\phi_{iab} = \partial_i \psi_{ab} \f$, and the operator
+ * \f$D := \partial_0 - N^k \partial_k \f$. The time derivative of N is then:
+ * \f{align*}
+ *  N^2/2 n^a n^b \pi_{ab} - N N^i/2 n^a n^b \phi_{iab}
+ *  =& N^2/2 n^a n^b n^c \partial_c \psi_{ab} - N N^i/2 (-(2/N) \partial_i N) \\
+ *  =& N^2/2 [-1/N^3 D[g_{jk} N^j N^k - N^2] - (N^j N^k / N^3)D[g_{jk}]
+ *           + 2(N^j / N^3) D[g_{jk} N^k] + (2 / N^2)(N^i \partial_i N)] \\
+ *  =& 1/(2N) [-D[g_{jk}N^jN^k - N^2] - N^jN^k D[g_{jk}]
+ *            + 2N N^k\partial_k N + 2N^j D[g_{jk}N^k]] \\
+ *  =& D[N] + N^k\partial_k N \\
+ *  =& \partial_0 N
+ * \f}
+ * where the simplification done for \f$\partial_i N\f$ is used to substitute
+ * for the second term (\f$N N^i/2 n^a n^b \phi_{iab}\f$).
+ *
+ * ==> \f$ \partial_0 N = (N/2)(N n^a n^b \pi_{ab} - N^i n^a n^b \phi_{iab}) \f$
+ */
+template <size_t SpatialDim, typename Frame, typename DataType>
+Scalar<DataType> time_deriv_of_lapse(
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept;
+
+/*!
+ * \ingroup GeneralRelativityGroup
+ * \brief Computes spatial derivatives of the shift vector from
+ *        the generalized harmonic and geometric variables
+ *
+ * \details Let the generalized harmonic spatial derivative variable be
+ * \f$\phi_{iab} = \partial_i \psi_{ab} \f$. The spatial derivatives of shift
+ * vector \f$N^i\f$ are:
+ * \f{align*}
+ *  \psi^{ja} N n^b \phi_{iab} - 2 n^j \partial_i N
+ *  =& (g^{ja} - n^j n^a) N n^b \phi_{iab} - 2 n^j \partial_i N \\
+ *  =& N g^{ja} n^b \partial_i \psi_{ab}
+ * \f}
+ * where we used the equation from above for \f$\partial_i N\f$. Then
+ * \f{align*}
+ * \psi^{ja} N n^b \phi_{iab} - 2 n^j \partial_i N
+ *  =& g^{ja} (\partial_i \psi_{a0} - N^k \partial _i \psi_{ak}) \\
+ *  =& g^{jl} (\partial_i N_l - N^k \partial_i g_{lk}) (\because g^{j0} = 0) \\
+ *  =& g^{jl} (N^k \partial_i g_{lk}
+ *             + g_{kl}\partial_i N^k - N^k \partial_i g_{kl}) \\
+ *  =& g^{jl} g_{kl} \partial_i N^k \\
+ *  =& \partial_i N^j
+ * \f}
+ * I.E.
+ * \f{align*}
+ *  \partial_i N^j =& N\psi^{ja} n^b \phi_{iab} - 2 n^j \partial_i N \\
+ *                 =& N (\psi^{ja} + n^j n^a) n^b \phi_{iab}.
```

Comment:
a -> k

Reply:
Functions computing derivs of 3plus1 quantities have been moved to  #1176 , #1177 , #1179 . Please review there.